### PR TITLE
LPD-31974 - ERC Support for specifications (specification labels)

### PR DESCRIPTION
**Pull request by matyaswollner**
https://liferay.atlassian.net/browse/LPD-31974

### CODE REVIEW

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:sf

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/SpecificationResourceTest.java [hunk 1]
```diff
@@ -6,15 +6,189 @@
...
+		specificationResource.patchSpecificationByExternalReferenceCode(
+			postSpecification.getExternalReferenceCode(),
+			randomPatchSpecification);
+
+		Specification expectedSpecification = postSpecification.clone();
+
+		BeanTestUtil.copyProperties(
+			randomPatchSpecification, expectedSpecification);
+
+		Specification getSpecification = specificationResource.getSpecification(
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/SpecificationResourceTest.java:111]: here use the getSpecificationByExternalReferenceCode

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by matyaswollner** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPSpecificationOptionLocalServiceImpl.java [hunk 2]
```diff
@@ -235,16 +235,21 @@ public CPSpecificationOption updateCPOptionCategoryId(
 	@Indexable(type = IndexableType.REINDEX)
 	@Override
 	public CPSpecificationOption updateCPSpecificationOption(
-			long cpSpecificationOptionId, long cpOptionCategoryId,
-			long listTypeDefinitionId, Map<Locale, String> titleMap,
-			Map<Locale, String> descriptionMap, boolean facetable, String key,
-			double priority, ServiceContext serviceContext)
+			String externalReferenceCode, long cpSpecificationOptionId,
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPSpecificationOptionLocalServiceImpl.java:238]: I don't know if we want to modify this method or add the addOrUpdate. I can find this solution also in other places (like [ListTypeDefinitionLocalServiceImpl](https://github.com/liferay/liferay-portal/blob/c3c6410fdd9089318054c83c4b2aa7cae0228077/modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeDefinitionLocalServiceImpl.java#L142) or [ListTypeEntryLocalServiceImpl](https://github.com/liferay/liferay-portal/blob/97db3371b6846eb9bd39f3102dcf267586511235/modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java#L188) or [CommerceOrderTypeLocalServiceImpl](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderTypeLocalServiceImpl.java#L240)). @andrea-ale-sbarra what do you think?

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:relevant

**Review by smottal — APPROVED** [review]

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:relevant

**Review by smottal — APPROVED** [review]

**Pull request closed without merging**